unittests/test-contracts/params_test/params_test.cpp

```diff
@@ -0,0 +1,238 @@
+
+      auto size = get_parameters_packed(packed.c_str(), packed.size(), buffer, sizeof(buffer));
+      ASSERT_LESS(size, sizeof(buffer));
+      return params_object{{buffer, size}};
+   }
+   size_t get_size() const{
+      return get_parameters_packed(packed.c_str(), packed.size(), 0, 0);
+   }
+};
+
+class [[eosio::contract("params_test")]] params_test : public eosio::contract {
+public:
+   using eosio::contract::contract;
+
+   [[eosio::action]] void maintest(){
+
+      //make sure no throw for zero parameters provided
+      params_object(0_ui).set();
+      ASSERT_EQ(params_object(0_ui).get(),
+                params_object(0_ui));
+
+      //1 for size, 1 for id, 8 for data
+      ASSERT_EQ(params_object({1_ui, 0_ui}).get_size(), 10);
+      //1 for size, 1 for id, 4 for data
+      ASSERT_EQ(params_object({1_ui, 1_ui}).get_size(), 6);
+      //1 for size, 1 for id, 2 for data
+      ASSERT_EQ(params_object({1_ui, 16_ui}).get_size(), 4);
+      //1 for size, 3 for ids, 8+4+2 for data
+      ASSERT_EQ(params_object({3_ui, 0_ui, 1_ui, 16_ui}).get_size(), 18);
+
+      blockchain_parameters old_way_params;
+      params_object(1_ui)(0_ui)(1024*1024*2_64).set();
+      get_blockchain_parameters(old_way_params);
+      ASSERT_EQ(params_object({1_ui,0_ui}).get(), 
+                params_object(1_ui)(0_ui)(1024*1024*2_64));
+      ASSERT_EQ(old_way_params.max_block_net_usage, 1024*1024*2);
+
+      params_object(1_ui)(1_ui)(20*100_32).set();
+      get_blockchain_parameters(old_way_params);
+      ASSERT_EQ(params_object({1_ui,1_ui}).get(), 
+                params_object(1_ui)(1_ui)(20*100_32));
+      ASSERT_EQ(old_way_params.target_block_net_usage_pct, 20*100);
+      
+      params_object(1_ui)(16_ui)(8_16).set();
+      get_blockchain_parameters(old_way_params);
+      ASSERT_EQ(params_object({1_ui,16_ui}).get(), 
+                params_object(1_ui)(16_ui)(8_16));
+      ASSERT_EQ(old_way_params.max_authority_depth, 8);
+
+      params_object(3_ui)(0_ui)(1024*1024*3_64)
+                         (1_ui)(30*100_32)
+                         (16_ui)(10_16).set();
+      get_blockchain_parameters(old_way_params);
+      ASSERT_EQ(params_object({3_ui, 0_ui, 1_ui, 16_ui}).get(), 
+                params_object(3_ui)(0_ui)(1024*1024*3_64)
+                                   (1_ui)(30*100_32)
+                                   (16_ui)(10_16));
+      ASSERT_EQ(old_way_params.max_block_net_usage, 1024*1024*3);
+      ASSERT_EQ(old_way_params.target_block_net_usage_pct, 30*100);
+      ASSERT_EQ(old_way_params.max_authority_depth, 10);
+
+      params_object(17_ui) (0_ui)(1024*1024*4_64)
+                           (1_ui)(35*100_32)
+                           (2_ui)(1024*1024*2_32)
+                           (3_ui)(1024_32)
+                           (4_ui)(700_32)
+                           (5_ui)(50_32)
+                           (6_ui)(120_32)
+                           (7_ui)(250'000_32)
```

Comment:
Did you determine if these are encoded correctly and just shown weird by Github?

Reply:
yes, it just weird Github highlight.
I checked, this is regular single quote:
Char | Dec | Hex | Oct
-- | -- | -- | --
' | 39 | 27 | 47